Drive Train and Shooter Subsystems

RaidingForPants opened the pull request:
Added the drive train and shooter subsystems. Also fixed the error Chris made when he gave two constants the same name.

RaidingForPants committed:
Merge remote-tracking branch 'refs/remotes/Team4169/master'

RaidingForPants committed:
Started the drive train

Made a header file

RaidingForPants committed:
Merge remote-tracking branch 'refs/remotes/Team4169/master' into Drive-Train

RaidingForPants committed:
Drive Train Finished

The drive train is finished

RaidingForPants committed:
Work In Progress-Shooter

RaidingForPants committed:
Shooter almost done

Need to use the constants from RobotMap

RaidingForPants committed:
Drive Train fix

Fixed the Drive Train to use a pointer

RaidingForPants committed:
Merge remote-tracking branch 'refs/remotes/Team4169/master'

RaidingForPants committed:
Merge remote-tracking branch 'refs/remotes/origin/Drive-Train'

RaidingForPants committed:
Comments for some reason

RaidingForPants committed:
Added ampersand

catvec commented:
[COMPLETE] Can you delete the ExampleSubsystem class

catvec commented:
[COMPLETE] Also rename the DriveTrain subsystem to Drivetrain(one word)

RaidingForPants committed:
Name changes

Renamed DriveTrain to Drivetrain, used the constants, and renamed the
shooter method

RaidingForPants committed:
Name Change

DriveTrain->Drivetrain

catvec commented on src/Subsystems/DriveTrain.cpp:
```diff
@@ -0,0 +1,23 @@
+#include <Subsystems/Drivetrain.h>
+#include "../RobotMap.h"
+
+Drivetrain::Drivetrain(RobotDrive drive): Subsystem("DriveTrain") {
```
Use correct name in `Subsystem` super call:

```
Drivetrain::Drivetrain(RobotDrive drive): Subsystem("Drivetrain") {
```

catvec commented on src/Subsystems/Shooter.cpp:
```diff
@@ -0,0 +1,30 @@
+#include "../RobotMap.h"
+
+Shooter::Shooter(): Subsystem("Shooter") {
+	lMotor=new Talon(LEFT_SHOOTER_MOTOR);
```
Space between equal sign

RaidingForPants committed:
Minor changes

ROODAY commented on src/Subsystems/DriveTrain.h:
```diff
@@ -0,0 +1,26 @@
+	void InitDefaultCommand();
+	void drive(float lSpeed, float rSpeed);
+
+};
```
Remove semicolon after class declaration

ROODAY commented on src/Subsystems/Shooter.h:
```diff
@@ -0,0 +1,28 @@
+	void run_shooter();
+	~Shooter();
+
+};
```
Remove semicolon after class declaration

catvec commented:
Can you remove the copyright headers from all your files?

catvec commented on src/Subsystems/DriveTrain.cpp:
```diff
@@ -0,0 +1,23 @@
+#include "../RobotMap.h"
+
+Drivetrain::Drivetrain(RobotDrive drive): Subsystem("Drivetrain") {
+	robot = &drive;
```
Make the constructor take a reference to a `RobotDrive` instead of creating a reference in the constructor:

```
Drivetrain::Drivetrain(Robotdrive *drive): 
```

RaidingForPants committed:
Fixed issues

Removed copyright header, changed the Drivetrain constructor parameter,
and removed the ExampleSubsytem objects from CommandBase

RaidingForPants committed:
Top blank line removal

RaidingForPants committed:
merged from upstream the long way because gh desktop stucks
